prov/util/src/util_coll.c

```diff
@@ -520,23 +521,17 @@ void util_coll_join_comp(struct util_coll_mc *coll_mc,
 void util_coll_barrier_comp(struct util_coll_mc *coll_mc,
 			    struct util_coll_comp_item *comp)
 {
-/* TODO: We should write a completion to the CQ, not the EQ
-	struct fi_eq_err_entry entry;
 	struct util_ep *ep;
-	ssize_t bytes;
 
-	memset(&entry, 0, sizeof(entry));
-	entry.fid = &coll_mc->mc_fid.fid;
-	entry.context = coll_mc->mc_fid.fid.context;
-	bytes = sizeof(struct fi_eq_entry);
-	free(comp->data);
+	ep = container_of(coll_mc->ep, struct util_ep, ep_fid);
 
-	ep  = container_of(coll_mc->ep, struct util_ep, ep_fid);
-	if (ofi_eq_write(&ep->eq->eq_fid, FI_COLL,
-			 &entry, (size_t) bytes, FI_COLLECTIVE) < 0)
+	if (ofi_cq_write(ep->tx_cq, comp->op_context, FI_COLLECTIVE,
+			 sizeof(struct util_coll_comp_data), comp->data, 0,
+			 comp->hdr.tag)) {
 		FI_WARN(ep->domain->fabric->prov, FI_LOG_DOMAIN,
-			"barrier collective - eq write failed\n");
-*/
+			"barrier collective - cq write failed\n");
+	}
+	free(comp->data);
```

Comment:
There's no great way to comment on the commit messages, but the message is too terse.  As a nit, the preferred way of labeling commits would be:
```
provider/topic: Answer this commit will do what here
```
This is a good source describing commit message format:

https://chris.beams.io/posts/git-commit/